--- linux-3.18.23/drivers/w1/masters/w1-gpio.c	2015-10-29 07:49:46.000000000 +0500
+++ b/drivers/w1/masters/w1-gpio.c	2016-02-24 10:00:43.834213998 +0500
@@ -87,7 +87,7 @@
 
 	if (of_get_property(np, "linux,open-drain", NULL))
 		pdata->is_open_drain = 1;
-
+    if (pdev->dev.of_node) {        /*(Virtual) of_node can be absent*/
 	gpio = of_get_gpio(np, 0);
 	if (gpio < 0) {
 		if (gpio != -EPROBE_DEFER)
@@ -106,7 +106,7 @@
 	pdata->ext_pullup_enable_pin = gpio;
 
 	pdev->dev.platform_data = pdata;
-
+    }
 	return 0;
 }
 
